waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.chnames = []
+
+        self.master_ome_tiff = self._get_master_ome_tiff(folder)
+        print(self.master_ome_tiff)
+        self._set_mm_meta()
+
+    def _set_mm_meta(self):
+        with TiffFile(self.master_ome_tiff) as tif:
+            self.mm_meta = tif.micromanager_metadata
+
+            if self.mm_meta['Summary']['Positions'] > 1:
+                self.stage_positions = []
+                for p in range(self.mm_meta['Summary']['Positions']):
+                    print(f"appending {self.mm_meta['Summary']['StagePositions'][p]}")
+                    self.stage_positions.append(self.mm_meta['Summary']['StagePositions'][p])
+
+            for ch in self.mm_meta['Summary']['ChNames']:
+                print(f"appending {ch}")
+                self.chnames.append(ch)
+
+            self.height = self.mm_meta['Summary']['Height']
+            self.width = self.mm_meta['Summary']['Width']
+            self.frames = self.mm_meta['Summary']['Frames']
+            self.slices = self.mm_meta['Summary']['Slices']
+            self.channels = self.mm_meta['Summary']['Channels']
+
+    def _extract_data(self, master_ome):
+        """
+        extract all series from ome-tiff and place into dict of (pos: zarr)
+        :param master_ome: full path to master OME-tiff
+        :return:
+        """
+        with TiffFile(master_ome) as tif:
+            for idx, tiffpageseries in enumerate(tif.series):
+                self._positions[idx] = zarr.open(tiffpageseries.aszarr(), mode='r')
+
+    def _get_master_ome_tiff(self, folder_):
```

Comment:
On dataset `/gpfs/CompMicro/rawdata/falcon/Ivan/20210128 HEK CAAX SiRActin/FOV1_1` this search incorrectly identifies `FOV1_1_MMStack_Default_23.ome.tif` as the master ome.tif file. I also get `invalid MicroManager display header` and `invalid MicroManager comments header` warnings. The code then proceeds to print "appending ... channels"

Reply:
I will explore this.  I believe in older versions of micro-manager, every file associated with the ome-tiff had the master scene info, so it didn't matter which one the class selects.  For time series, this might be true too (?).

The warnings are probably output from tifffile.  It seems to care about json formatting.  If this is the case, the proper place to fix it is within micro-manager.

Reply:
>I believe in older versions of micro-manager, every file associated with the ome-tiff had the master scene info, so it didn't matter which one this reader selects. For time series, this might be true too (?).

I believe the version I'm using is from ~June last year. I can easily switch to a newer version, I'm not particularly tied to it.

> It seems to care about json formatting. If this is the case, the proper place to fix it is within micro-manager.

I recently learned how to suppress certain warnings: https://docs.python.org/3/library/warnings.html. Either that, or we fix the issue in MM, as you said.

Reply:
>I recently learned how to suppress certain warnings: https://docs.python.org/3/library/warnings.html. Either that, or we fix the issue in MM, as you said.
The recent pushes to this reader have a flag to set logging level.  By default it will be set to logging.ERROR which suppresses warnings.

Reply:
On dataset `/gpfs/CompMicro/rawdata/falcon/Ivan/20210128 HEK CAAX SiRActin/FOV1_1` the `_get_master_ome_tiff` still incorrectly identifies `FOV1_1_MMStack_Default_23.ome.tif` as the master-ome file and the reader doesn't work. @bryantChhun can you take a look at this again?

Also, if I pass `/gpfs/CompMicro/rawdata/falcon/Ivan/20210108 HEK_CAAX_SiRActin` as the data path, the reader errors out with:
```
~/Documents/waveorder/waveorder/io/reader.py in _get_master_ome_tiff(self, folder_)
    149                     continue
    150 
--> 151         return os.path.join(folder_, ome_master)
    152 
    153     def get_zarr(self, position):

UnboundLocalError: local variable 'ome_master' referenced before assignment
```
`_get_master_ome_tiff` should not error out if we pass a folder that doesn't contain MM data or if `ome_master` can't be found

Reply:
I'll take a look at this today.  Now knowing that the non-master files should contain a pointer to the master, I could refactor this to be more efficient.

Reply:
I have restructured the way master-files are found.  It now identifies it directly from the metadata without iteration.